capreolus/sampler/__init__.py

```diff
@@ -0,0 +1,97 @@
+import random
+import torch.utils.data
+
+from capreolus.utils.loginit import get_logger
+
+
+logger = get_logger(__name__)
+
+
+class TrainDataset(torch.utils.data.IterableDataset):
```

Comment:
@andrewyates Do you think this should be a module? I couldn't think of anything that the user might want to configure here, except maybe the seed.

Reply:
no, I think all the (minimal) config options should be passed in. The seed is used across modules, so it makes sense to keep it as a task config option